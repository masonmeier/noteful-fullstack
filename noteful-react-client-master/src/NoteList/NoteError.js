// Error Boundary component
// NOTE: MUST use class

import React from 'react';

class NoteError extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			hasError: false
		};
	}

	static getDerivedStateFromError(error) {
		// Update state so the next render will show the fallback UI.
		return { hasError: true };
	}

	componentDidCatch(error, errorInfo) {
		// You can also log the error to an error reporting service
		// logErrorToMyService(error, errorInfo);
	}

	render() {
		if (this.state.hasError) {
			// You can render any custom fallback UI
			return <h2>Could not display this note.</h2>;
		}
		return this.props.children;
	}
}
export default NoteError;
