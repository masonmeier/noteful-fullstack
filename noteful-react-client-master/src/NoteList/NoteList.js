import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import NoteError from './NoteError';
import NoteItem from '../NoteItem/NoteItem';
import NotefulContext from '../NotefulContext';

import config from '../config';

const NoteList = props => {
	// need to grab NotefulContext (globals)
	const contextType = useContext(NotefulContext);
	const { notes } = contextType;

	let folderId;

	// if selected a folder, show only the notes in that folder
	// otherwise show all notes from all folders
	if (config.DATASOURCE === 'postgresql') {
		folderId = parseInt(props.match.params.id_folder);
	} else {
		folderId = props.match.params.id_folder;
	}

	const foldernotes = props.match.params.id_folder
		? notes.filter(note => note.id_folder === folderId)
		: notes;

	return (
		<>
			<header>
				<h2>Notes</h2>
				&nbsp;&nbsp;
				<NavLink
					// if just passing url:
					// to={'/add-note'}

					// to pass selected id_folder:
					to={{
						pathname: '/add-note',
						state: { id_folder: props.match.params.id_folder }
					}}
				>
					<button className="btn-add">+</button>
				</NavLink>
			</header>
			{foldernotes.length > 0 ? (
				foldernotes.map(note => (
					<section key={note.id}>
						<NoteError>
							<div className="note">
								{/*
							{...note}
							use spread here IF want whole note object, and then get inside NoteItem via note.key
							alternative is
							note={note} and in NoteItem get via props.note.key

							NOTE:
							MUST pass {...props} to have the history, location, and match props inside NoteItem
							*/}
								<NoteItem note={note} {...props} />
							</div>
						</NoteError>
					</section>
				))
			) : (
				<section className="no-border">
					<div className="note">No notes in this folder.</div>
				</section>
			)}
		</>
	);
};

export default NoteList;

// to catch bugs
// check that get a notes array that has id, name, and modified
// this array is the "notes" variable coming from context
NoteList.propTypes = {
	notes: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string.isRequired,
			name: PropTypes.string.isRequired,
			modified: PropTypes.instanceOf(Date).isRequired
		})
	)
};
