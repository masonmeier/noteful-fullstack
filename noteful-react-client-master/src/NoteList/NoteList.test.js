import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import NoteList from './NoteList';

describe(`NoteList component`, () => {
	const props = {
		match: {
			params: {
				id_folder: '101'
			}
		},
		notes: [
			{
				id: '1',
				id_folder: '101',
				name: 'test-note-title',
				modified: new Date(2019, 8, 30)
			}
		]
	};

	it('renders a NoteList by default', () => {
		const wrapper = shallow(<NoteList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the NoteList given props', () => {
		const wrapper = shallow(<NoteList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
