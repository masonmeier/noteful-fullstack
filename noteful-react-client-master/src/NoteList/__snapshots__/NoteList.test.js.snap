// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList component renders a NoteList by default 1`] = `
<Fragment>
  <header>
    <h2>
      Notes
    </h2>
      
    <NavLink
      to={
        Object {
          "pathname": "/add-note",
          "state": Object {
            "id_folder": "101",
          },
        }
      }
    >
      <button
        className="btn-add"
      >
        +
      </button>
    </NavLink>
  </header>
  <section
    className="no-border"
  >
    <div
      className="note"
    >
      No notes in this folder.
    </div>
  </section>
</Fragment>
`;

exports[`NoteList component renders the NoteList given props 1`] = `
<Fragment>
  <header>
    <h2>
      Notes
    </h2>
      
    <NavLink
      to={
        Object {
          "pathname": "/add-note",
          "state": Object {
            "id_folder": "101",
          },
        }
      }
    >
      <button
        className="btn-add"
      >
        +
      </button>
    </NavLink>
  </header>
  <section
    className="no-border"
  >
    <div
      className="note"
    >
      No notes in this folder.
    </div>
  </section>
</Fragment>
`;
